mobilecoind/strategies/burn.py

```diff
@@ -0,0 +1,203 @@
+            sender_monitor_id=monitor_id,
+            change_subaddress=change_subaddress,
+            input_list=input_list,
+            burn_amount=burn_amount,
+            fee=fee,
+            tombstone=tombstone,
+            token_id=token_id,
+            redemption_memo=redemption_memo,
+            enable_destination_memo=enable_destination_memo,
+        )
+        return self.stub.GenerateBurnRedemptionTx(request).tx_proposal
+
+    def submit_tx(self, tx_proposal):
+        return self.stub.SubmitTx(
+            SubmitTxRequest(
+                tx_proposal=tx_proposal,
+            )
+        )
+
+    def wait_for_monitor_to_sync(self, monitor_id, poll_interval=0.1):
+        while True:
+            network_block_index = self.get_network_block_index()
+            status = self.stub.GetMonitorStatus(
+                GetMonitorStatusRequest(
+                    monitor_id=monitor_id
+                )
+            ).status
+            if status.next_block >= network_block_index:
+                break
+            time.sleep(poll_interval)
+
+
+if __name__ == '__main__':
+    args = parse_args()
+    logging.debug(args)
+
+    client = MobilecoindClient(f"{args.mobilecoind_host}:{args.mobilecoind_port}")
+    account_data = client.new_monitor_from_keyfile(args.key)
+    monitor = account_data.monitor_id
+
+    client.wait_for_monitor_to_sync(monitor)
+
+    balance = client.get_balance(monitor, args.token_id)
+
+    if balance < args.value + args.fee:
+        raise Exception(f"Insufficient balance: {balance} < {args.value} + {args.fee}")
+
+    utxos = client.get_utxos(monitor, args.token_id)
+    tx_proposal = client.generate_burn_redemption_tx(
+        monitor,
+        utxos,
+        args.value,
+        args.fee,
+        args.token_id,
+    )
+    logging.info("Constructed burn transaction")
+    logging.debug(tx_proposal)
+
+    # Search for the burn txout. mobilecoind will create one "outlay" which is
+    # the burn, and there is a map in the TxProposal that tells you which TxOut it is.
+    if len(tx_proposal.outlay_index_to_tx_out_index) != 1:
+        logging.warning(f"Unexpected number of outlays: {tx_proposal.outlay_index_to_tx_out_index}")
+    tx_out_public_key_bytes = tx_proposal.tx.prefix.outputs[tx_proposal.outlay_index_to_tx_out_index[0]].public_key.data
+    tx_out_public_key_hex = tx_out_public_key_bytes.hex()
+    logging.info(f"*** The hex bytes of burn TxOut public key are: {tx_out_public_key_hex}")
+
+    logging.info("Submitting burn transaction")
+    submit_response = client.submit_tx(tx_proposal)
+    print(submit_response)
+
+    time.sleep(3)
+
+    client.wait_for_monitor_to_sync(monitor)
+    new_balance = client.get_balance(monitor, args.token_id)
+
+    if new_balance == balance - args.value - args.fee:
+        logging.info("Burn transaction was successful")
+    else:
+        logging.error("Burn transaction appears to be unsuccessful")
```

Comment:
Consider exiting with a non-zero status
```suggestion
        logging.error("Burn transaction appears to be unsuccessful")
        sys.exit(1)
```